extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGroupHandler.java

```diff
@@ -0,0 +1,191 @@
+            coapClient.setEndpoint(handler.groups.getEndpoint());
+        } catch (URISyntaxException e) {
+            logger.debug("Illegal device URI `{}`: {}", uriString, e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            return;
+        }
+        updateStatus(ThingStatus.UNKNOWN);
+        active = true;
+
+        scheduler.schedule(() -> {
+            coapClient.startObserve(this);
+        }, 3, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public synchronized void dispose() {
+        active = false;
+        if (coapClient != null) {
+            coapClient.shutdown();
+        }
+        super.dispose();
+    }
+
+    @Override
+    public void setStatus(ThingStatus status, ThingStatusDetail statusDetail) {
+        if (active && getBridge().getStatus() != ThingStatus.OFFLINE && status != ThingStatus.ONLINE) {
+            updateStatus(status, statusDetail);
+            // we are offline and lost our observe relation - let's try to establish the connection in 10 seconds again
+            scheduler.schedule(() -> {
+                coapClient.startObserve(this);
+            }, 10, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void onUpdate(JsonElement data) {
+        if (active && !(data.isJsonNull())) {
+            state = new GroupData(data);
+            updateStatus(ThingStatus.ONLINE);
+
+            if (state.getOnOffState()) {
+                updateState(CHANNEL_GROUP_STATE, OnOffType.ON);
+            } else {
+                updateState(CHANNEL_GROUP_STATE, OnOffType.OFF);
+            }
+        }
+    }
+
+    private void set(String payload) {
+        logger.debug("Sending payload: {}", payload);
+        coapClient.asyncPut(payload, this);
+    }
+
+    private void setState(OnOffType onOff) {
+        GroupData data = new GroupData();
+        data.setOnOffState(onOff == OnOffType.ON);
+        set(data.getJsonString());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            logger.debug("Refreshing channel {}", channelUID);
+            coapClient.asyncGet(this);
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case CHANNEL_GROUP_STATE:
+                handleGroupStateCommand(command);
+                break;
+
+            default:
+                logger.error("Unknown channel UID {}", channelUID);
+        }
+    }
+
+    private void handleGroupStateCommand(Command command) {
+        if (command instanceof OnOffType) {
+            setState(((OnOffType) command));
+        } else {
+            logger.debug("Cannot handle command {} for channel {}", command, CHANNEL_GROUP_STATE);
+        }
+    }
+
+    /**
+     * This class is a Java wrapper for the raw JSON data about the group state.
+     */
+    private static class GroupData {
```

Comment:
Here again, my feeling from just looking at the code is that `TradfriLightHandler.LightData` could be reused or at least should be made reusable.